### ISSUE TITLE
segfault using --score-N option for ambiguous nucleotide score

### ISSUE THREAD

**Issue by nrizzo** [lh3/minimap2#1226]
Hi! I am testing minimap2 (69e3629) with base alignment on a hard masked reference and trying to penalize the matching of `N`s. I discovered that choosing `--score-N` values of 13 or higher results in a segfault:
```
./minimap2 --score-N 13 -a test/MT-human.fa test/MT-orang.fa > test.sam
[M::mm_idx_gen::0.005*0.96] collected minimizers
[M::mm_idx_gen::0.007*0.81] sorted minimizers
[M::main::0.007*0.81] loaded/built the index for 1 target sequence(s)
[M::mm_mapopt_update::0.007*0.82] mid_occ = 10
[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1
[M::mm_idx_stat::0.007*0.82] distinct minimizers: 3111 (100.00% are singletons); average occurrences: 1.000; average spacing: 5.326; total length: 16569
Segmentation fault (core dumped)
```

Also, it is unclear to me if `--score-N` defines a bonus or a penalty from reading the man page and [this closed issue](https://github.com/lh3/minimap2/issues/484), but looking at the [source code](https://github.com/lh3/minimap2/blob/69e36299168d739dded1c5549f662793af10da83/align.c#L14) it seems that positive `--score-N` values get turned into negative values, so it is *always* a score penalty, right? Also, the score matrix accepts int8 values, so scores should only be in the range [-128,127]! Can you confirm this? Thank you in advance.

Best,
~Nicola

**Comment by lh3** [comment]
You are right on all points. I don't know why a large --score-N would cause the segfault, but this parameter is supposed to be smaller than mismatch penalty. I added a check.
